### ISSUE TITLE
Bug and Enhancement request of replace function.

### ISSUE THREAD

**Issue by toshiakiasakura** [man-group/dtale#671]
**Current Behavior:**
- Open jupyter lab
- read df with `pd.read_csv`
- oepn dtale by `dtale.show()`
- Select a column, and select **Replacements**
- Select replacement type as **Contains Char/Substring** 
- Fill Search for and replace with 
- Push Replace button

However, this can not replace cells with inclusion of search for characters.

The current auto-generated code is like this.
```python
import re

regex_pat = re.compile(r'^ *' + re.escape('Kraeff') + ' *$', flags=re.UNICODE)
df.loc[:, 'Name'] = df['Name'].replace(regex_pat, 'AAAA', regex=True)
```
However, regex_pat part of line, should be like this. 
```python
regex_pat = re.compile(r'^.*' + re.escape('Kraeff') + '.*$', flags=re.UNICODE)
```
After replacing space with dot, it works well. 

Also, I want partial replacement function, which can be achieved by `df["name"].str.replace("hoge", "non")`
Without this type of replacement functions, we can not confotably use dtale for data cleaning purposes.

I am very fascinated with this open source package, so that I hope the suggested bug will be fixed and function will be implemented.

**Comment by aschonfeld** [comment]
@toshiakiasakura great suggestion! I'll work on adding these fixes/features and let you know when a new version is ready. Thanks

**Comment by aschonfeld** [comment]
@toshiakiasakura I did some more digging and I forgot that the partial replacement is available you just need to go to "Dataframe Functions" in the main menu and then select the column you want and the replacement you want made.  Also if you don't want to create a new column select "Inplace" 
<img width="1117" alt="image" src="https://user-images.githubusercontent.com/11547371/173876713-d243a891-3f6d-434b-a5bd-59bc5beca1e3.png">

If you want I can look into adding this function to the "Replacements" popup

**Comment by toshiakiasakura** [comment]
Thank you very much! 
I think if the partial replacement function is also built in the "Replacements" popup, it becomes more user friendly, and I like that way. It will be consistent, and we can easily find functions.

**Comment by aschonfeld** [comment]
@toshiakiasakura just released v2.6.0 with these enhancements/updates
